app/store/networkDrawer.ts

```diff
@@ -0,0 +1,30 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+export interface NetworkDrawerState {
+  isLoading: boolean
+  isOpen: boolean
+  height: number
+  title: string
+  txid?: string
```

Comment:
I don't think you should not track component state in store (UI state), you should only track data

In this case, pending transaction data

Reply:
Instead of using store for UI state, we need to pass status of transaction? then we just derive the UI state from that. Am I correct?

Reply:
Ideally, the scenario would be:

1. push signed transaction to the drawer (global state)
2. drawer will enqueue the transaction and send the whale API (if online)
3. drawer will spin loop until and transaction gets confirmed therefore removing it from the drawer

Something like this: 
```ts
interface State {
  transactions: {
    broadcasted: boolean
    signed: CTransactionSegWit[]
  }[]
}
```

Reply:
With this approach, I feel like we're making it coupled with transactions. Wouldn't it be better to make it a dumb component and let the called component dictate when/what to display this component? So it will be flexible enough to use on other use cases as well. (e.g, fetching data, displaying errors etc.)

Reply:
If that's the case, the transactions.signed will behaves like a queue. The drawer will handle the all the submission, status, error rendering (within drawer component)?

Reply:
Yeah it should behave like a queue for the blockchain. The entire idea of that component is to add as a middleware for connecting to the blockchain.

Be it network status or bridging transactions.